src/Ldap.php

```diff
@@ -923,17 +923,16 @@ public function search($filter, $basedn = null, $scope = self::SEARCH_SCOPE_SUB,
         if ($search === false) {
             throw new Exception\LdapException($this, 'searching: ' . $filter);
         }
+
+        $iterator = new Collection\SortableIterator($this, $search);
+
         if ($sort !== null && is_string($sort)) {
-            ErrorHandler::start(E_WARNING);
-            $isSorted = ldap_sort($resource, $search, $sort);
-            ErrorHandler::stop();
+            $isSorted = $iterator->sort($sort);
             if ($isSorted === false) {
                 throw new Exception\LdapException($this, 'sorting: ' . $sort);
```

Comment:
It's an artifact of the current implementation. If I move it to the sort() method (which in my eyes makes much more sense) I would change the return value of the sort method from ```boolean``` to ```self``` (or whatever is appropriate in that file).